homeassistant/components/emoncms/sensor.py

```diff
@@ -107,6 +106,10 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
         if sensor_names is not None:
             name = sensor_names.get(int(elem["id"]), None)
 
+        unit_of_measurement = elem.get("unit", "")
+        if unit_of_measurement == "":
```

Comment:
Here we could use the `empty string` and `None` as a falsy value:
```suggestion
        if not unit:
```